numba/cuda/compiler.py

```diff
@@ -786,6 +787,16 @@ def __init__(self, dispatcher, griddim, blockdim, stream, sharedmem):
         self.stream = stream
         self.sharedmem = sharedmem
 
+        num_blocks = griddim[0] * griddim[1] * griddim[2]
+        ctx = get_context()
+        smcount = ctx.device.MULTIPROCESSOR_COUNT
+
+        if (config.LOW_OCCUPANCY_WARNINGS) and (num_blocks < 2 * smcount):
+            msg = ("Number of blocks ({blocks}) that are  < 2 * SM ({sm}) "
```

Comment:
I think `will generate inefficient kernel code` is a bit confusing - how about phrasing the warning as `Grid size (<size>) less than 2 * SM count (2 * (<SM count>) = (<SM count * 2>) will likely result in GPU underutilization due to low occupancy`?